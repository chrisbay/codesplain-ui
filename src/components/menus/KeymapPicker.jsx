import React, { PropTypes } from 'react';
import memoize from 'lodash/memoize';
import { MenuItem } from 'material-ui';
import KeyboardIcon from 'material-ui/svg-icons/hardware/keyboard';
import ArrowDropRight from 'material-ui/svg-icons/navigation-arrow-drop-right';

const keymapOptions = [
  ['Default', 'default'],
  ['Emacs', 'emacs'],
  ['Vim', 'vim'],
  ['Sublime Text', 'sublime'],
];

const makeKeybindingOptions = memoize((selected, onChange) => (
  keymapOptions.map(([primaryText, value]) => (
    <MenuItem
      checked={value === selected}
      key={value}
      onTouchTap={() => { onChange(value); }}
      primaryText={primaryText}
      value={value}
    />
  ))
));

const KeymapPicker = (props) => {
  const {
    selectedKeymap,
    onChange,
  } = props;
  return (
    <MenuItem
      leftIcon={<KeyboardIcon />}
      menuItems={makeKeybindingOptions(selectedKeymap, onChange)}
      onChange={onChange}
      primaryText="Keymap"
      rightIcon={<ArrowDropRight />}
    />
  );
};

KeymapPicker.propTypes = {
  onChange: PropTypes.func.isRequired,
  selectedKeymap: PropTypes.string,
};

KeymapPicker.defaultProps = {
  selectedKeymap: 'default',
};

export default KeymapPicker;
