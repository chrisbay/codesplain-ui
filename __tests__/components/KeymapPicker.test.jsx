import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import KeymapPicker from '../../src/components/menus/KeymapPicker';

const defaultProps = {
  onChange: jest.fn(),
};

describe('<KeyPicker />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshot tests', () => {
    it('matches default snapshot', () => {
      const wrapper = shallowWithContext(<KeymapPicker {...defaultProps} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
