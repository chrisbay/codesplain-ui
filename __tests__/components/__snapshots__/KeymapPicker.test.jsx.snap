exports[`<KeyPicker /> snapshot tests matches default snapshot 1`] = `
<MenuItem
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  checked={false}
  desktop={false}
  disabled={false}
  focusState="none"
  insetChildren={false}
  leftIcon={<HardwareKeyboard />}
  menuItems={
    Array [
      <MenuItem
        anchorOrigin={
            Object {
                "horizontal": "right",
                "vertical": "top",
              }
        }
        checked={true}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        primaryText="Default"
        value="default" />,
      <MenuItem
        anchorOrigin={
            Object {
                "horizontal": "right",
                "vertical": "top",
              }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        primaryText="Emacs"
        value="emacs" />,
      <MenuItem
        anchorOrigin={
            Object {
                "horizontal": "right",
                "vertical": "top",
              }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        primaryText="Vim"
        value="vim" />,
      <MenuItem
        anchorOrigin={
            Object {
                "horizontal": "right",
                "vertical": "top",
              }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        primaryText="Sublime Text"
        value="sublime" />,
    ]
  }
  onChange={[Function]}
  primaryText="Keymap"
  rightIcon={<NavigationArrowDropRight />} />
`;
